### PR TITLE
Use AWS SDK v2 for SecretsManager

### LINKED ISSUE #74: AWS sdk version
Hello 👋 

The AWS sdk version is not using the latest one which is v2. 

You should migrate to this one because they have versioned their sdk by modules which allows for a smaller binary overall.

### PR DESCRIPTION
**Pull request by jefflinse**
Updates the AWS Secrets Manager provider to use AWS SDK V2 for fetching secrets. Resolves #74.

### CODE REVIEW

**Review by busser — APPROVED** [review]

**Review by busser — COMMENTED** [review]
End-to-end tests are failing on my machine. Attempting to enable them on this PR to troubleshoot

File: internal/whisper/providers/awssm/client.go [hunk 1]
```diff
@@ -6,24 +6,19 @@ import (
...
 // New returns a client that fetches secrets from AWS Secrets Manager.
 func New() (*client, error) {
-	session, err := session.NewSession()
-	if err != nil {
-		return nil, err
-	}
-
-	c := secretsmanager.New(session)
+	opts := secretsmanager.Options{}
+	c := secretsmanager.New(opts)
```
Comment (busser) [inline internal/whisper/providers/awssm/client.go:21]: Looks like passing empty options prevents the client from configuring itself based on the environment whisper is running in. More details in other comment.

**Review by busser — CHANGES_REQUESTED** [review]

File: internal/whisper/providers/awssm/client.go (hunk 1 above)
Comment (busser) [reply internal/whisper/providers/awssm/client.go:21]: LoadDefaultConfig docs: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#LoadDefaultConfig

NewFromConfig docs: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/secretsmanager#NewFromConfig

Comment (jefflinse) [reply internal/whisper/providers/awssm/client.go:21]: Yep, you're right. Fixing now.

Are you okay with a `context.TODO()` in the provider constructor or should I go ahead and plumb through a proper context from the resolver?

**Comment by jefflinse** [comment]
E2E tests for the provider are now passing for me:

```
whisper/providers/awssm $ go test -tags=e2e ./...
ok  	github.com/busser/whisper/internal/whisper/providers/awssm	0.329s
```

**Review by busser — DISMISSED** [review]
Awesome! Thank you for doing this 🙏

**Comment by busser** [comment]
Changes released in v0.4.1.

Thanks again @jefflinse !

**Pull request merged**